library/x509_crt.c

```diff
@@ -590,6 +590,74 @@ static int x509_get_ext_key_usage( unsigned char **p,
     return( 0 );
 }
 
+/*
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ * HardwareModuleName ::= SEQUENCE {
+ *                           hwType OBJECT IDENTIFIER,
+ *                           hwSerialNum OCTET STRING }
+ *
+ * NOTE: we only parse and use otherName of type HwModuleName,
+ * as defined in RFC 4108 at this point.
+ */
+static int x509_get_other_name( unsigned char **p,
+                                const unsigned char *end,
+                                mbedtls_x509_name *hardware_module_name )
```

Comment:
I think there is mismatch between the generic name of the function and the specific name of its `hardware_module_name` argument. However, given that it's `static` and can change at any time, it's not a blocker.

Reply:
Yes, I am aware. My intention was this function can be enhanced to support other "other names" in the future, but currently only hardware module name supported.
In addition, this function parses the `other name` SAN section, and returns only the hardware module name